source/extensions/filters/http/cache/http_cache.h

```diff
@@ -268,9 +269,17 @@ class HttpCacheFactory : public Config::TypedFactory {
   //
   // Pass factory context to allow HttpCache to use async client, stats scope
   // etc.
-  virtual std::shared_ptr<HttpCache>
+
+  // Get cache for global config
+  virtual HttpCachePtr
   getCache(const envoy::extensions::filters::http::cache::v3::CacheConfig& config,
            Server::Configuration::FactoryContext& context) PURE;
```

Comment:
Rather than splitting this into two identical functions, I think it would work to have this function take a [Server::Configuration::CommonFactoryContext&](https://github.com/envoyproxy/envoy/blob/main/envoy/server/factory_context.h#L108).

(Then you could also propagate this change up to where you have a templated getCache to simplify that too.)